nbconvert/templates/html/basic.tpl

```diff
@@ -155,6 +155,10 @@ height={{ height }}
 {%- if output | get_metadata('unconfined', 'image/png') %}
 class="unconfined"
 {%- endif %}
+{%- set alttext=(output | get_metadata('alt', 'image/png')) or (cell | get_metadata('alt')) -%}
```

Comment:
What if the cell has multiple images? Are we ok with all images getting the `cell | get_metadata('alt')`?

Reply:
If a cell has multiple image outputs, the alt-text should be added to the output's metadata, not the cell's metadata. But there's no way to edit an output's metadata using Jupyter notebook or Jupyter lab. We can only edit the cell's metadata. We can use `IPython.display.Image` to programmatically set the output metadata, but that doesn't help people creating figures with `matplotlib` because you can't set the output metadata through `matplotlib`, and even if you could, not everyone would want to type out the alt-text in their code like that.

People who want or need to add alt-text to their notebooks for accessibility reasons can limit their cell outputs to one figure per cell, which isn't much of a limitation since that's what a lot of people do anyway. And without this change, all notebooks have zero alt-text added to figures and images, making Jupyter notebook blog posts less accessible to the blind or low vision community.

Reply:
Yeah I think that's a fair compromise. Just wanted to make sure it was thought out.

Thanks for the improvement!

Reply:
You are welcome!

I'd also like to update the documentation somewhere so people know about this feature. Can you advise how I should go about that?  Which github project should I update and where should the documentation go?

Reply:
It's this repo -- you can find to root of the docs here: https://github.com/jupyter/nbconvert/tree/master/docs